classes/core/PKPRequest.php

```diff
@@ -127,7 +127,10 @@ public function redirectUrl($url)
             return;
         }
 
+        Application::get()->getRequest()->getSessionGuard()->sendCookies();
```

Comment:
It looks like this is necessary anywhere that the code terminates with a response to the user, right? I think there might be at least one more in the end of `PKPRouter::_authorizeInitializeAndCallRequest` where `JSONMessage` instances are processed.